devicetypes/smartthings/ozom-smart-siren.src/ozom-smart-siren.groovy

```diff
@@ -127,32 +146,51 @@ def configure() {
 
 def both() {
 	log.debug "both()"
-	on()
+	state.alarmCmd = ALARM_BOTH
+	def warningDuration = state.maxDuration ? state.maxDuration : DEFAULT_MAX_DURATION
+	state.lastDuration = warningDuration
+
+	zigbee.command(IAS_WD_CLUSTER, COMMAND_IAS_WD_START_WARNING, MODE_BOTH, DataType.pack(warningDuration, DataType.UINT16), STROBE_DUTY_CYCLE, STROBE_LEVEL)
 }
 
 def siren() {
 	log.debug "siren()"
-	on()
+
+	state.alarmCmd = ALARM_SIREN
+	def warningDuration = state.maxDuration ? state.maxDuration : DEFAULT_MAX_DURATION
+	state.lastDuration = warningDuration
+
+	// start warning, burglar mode, no strobe, siren very high
+	zigbee.command(IAS_WD_CLUSTER, COMMAND_IAS_WD_START_WARNING, MODE_SIREN, DataType.pack(warningDuration, DataType.UINT16), "00", "00")
 }
 
 def strobe() {
-	log.warn "strobe() is not supported by this device"
+	log.debug "strobe()"
+
+	state.alarmCmd = ALARM_STROBE
+	def warningDuration = state.maxDuration ? state.maxDuration : DEFAULT_MAX_DURATION
+	state.lastDuration = warningDuration
+
+	zigbee.command(IAS_WD_CLUSTER, COMMAND_IAS_WD_START_WARNING, MODE_STROBE, DataType.pack(warningDuration, DataType.UINT16), STROBE_DUTY_CYCLE, STROBE_LEVEL)
```

Comment:
there's a lot of duplicated code here

Reply:
made new commit to reduce duplicated codes